### DISCUSSION TITLE
Press button emulation and API

### DISCUSSION THREAD

**Discussion by Tonio16** [zwave-js/zwave-js-ui#2140] [Q&A]
Hello

I was using previously to zwavejs2mqtt a solution based on openzwave. It was the zwave plugin of jeedom. With it, it is possible to emulate  a Press Button. 
For example with the fibaro fgr-222 and 145/7 or 145/8, we can tilt venetian blind. It was also used fort the stop. 

I thought that the API of zwavejs2mqtt could help here but I didn't find the proper function to emulate a press. 
Is there a solution here or must I simply forget about it? 

TX

Antoine

**Comment by robertsLando** [comment]
I understand what you want and I think it could be possible using mqtt apis but I dunno what's the api needed in this case @AlCalzone could you help here?

**Reply by AlCalzone** [reply]
I'm not sure I understand what's being asked here. Tilt/Position of these shutters can be controlled by setting the corresponding values:
```js
	{
		commandClass: CommandClasses["Manufacturer Proprietary"],
		endpoint, // probably 0
		property: "fibaro",
		propertyKey: "venetianBlindsPosition",
	}
```
and
```js
	{
		commandClass: CommandClasses["Manufacturer Proprietary"],
		endpoint, // probably 0
		property: "fibaro",
		propertyKey: "venetianBlindsTilt",
	}
```

or by using these CC APIs:
https://zwave-js.github.io/node-zwave-js/#/api/CCs/ManufacturerProprietary

**Reply by Tonio16** [reply]
Hello

With a picture to illustrate what I was using before it should be easier to understand:
<a href="https://ibb.co/cL13MDm"><img src="https://i.ibb.co/DptWs5q/zwavejs-github.jpg" alt="zwavejs-github" border="0"></a>
Class = commandClass
Instance = ?
Index = endpoint index.

With this configuration and the use of index 7 or 8 instead of the 0 one, I can emulate a button release, use for the stop, or a push on upper or lower button, for the venitian blind.
It would be a nice to have for me and it will help to better understand the possibilities of zwavejs2mqtt.

Tx
Antoine

**Reply by AlCalzone** [reply]
Instance and index are OpenZWave specific. Instance roughly translates to the endpoint, but I think instance 1 is endpoint 0, instance 2 is endpoint 1 and so on...
Index has no direct translation, its a combination of property and propertyKey.

**Reply by robertsLando** [reply]
@Tonio16 Did you try to send values from control panel?

**Comment by Tonio16** [comment]
@robertsLando
No, I don't know how to process here. 

Antoine

**Reply by robertsLando** [reply]
@Tonio16 Open ui, control panel, go on the node you want to control expand it go to the Manufacturer Propetary CC and look for the `venetianBlindsTilt` and `venetianBlindsPosition` values, try to send a command to them and check if that is working

**Reply by Tonio16** [reply]
Ok, this one are working fine. 

Antoine

**Reply by robertsLando** [reply]
@Tonio16 ok so if you want to do the same using mqtt follow this: https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=set-values
